src/node/http.ts

```diff
@@ -1,885 +1,253 @@
-   * allow the user to authenticate only once. This will use the highest level
-   * domain (e.g. `coder.com` over `test.coder.com` if both are specified).
-   */
-  private getCookieDomain(host: string): string | undefined {
-    const idx = host.lastIndexOf(":")
-    host = idx !== -1 ? host.substring(0, idx) : host
-    if (
-      // Might be blank/missing, so there's nothing more to do.
-      !host ||
-      // IP addresses can't have subdomains so there's no value in setting the
-      // domain for them. Assume anything with a : is ipv6 (valid domain name
-      // characters are alphanumeric or dashes).
-      host.includes(":") ||
-      // Assume anything entirely numbers and dots is ipv4 (currently tlds
-      // cannot be entirely numbers).
-      !/[^0-9.]/.test(host) ||
-      // localhost subdomains don't seem to work at all (browser bug?).
-      host.endsWith(".localhost") ||
-      // It might be localhost (or an IP, see above) if it's a proxy and it
-      // isn't setting the host header to match the access domain.
-      host === "localhost"
-    ) {
-      logger.debug("no valid cookie doman", field("host", host))
-      return undefined
-    }
-
-    this.options.proxyDomains.forEach((domain) => {
-      if (host.endsWith(domain) && domain.length < host.length) {
-        host = domain
-      }
-    })
-
-    logger.debug("got cookie doman", field("host", host))
-    return host ? `Domain=${host}` : undefined
-  }
-
-  /**
-   * Return a response if the request should be proxied. Anything that ends in a
-   * proxy domain and has a *single* subdomain should be proxied. Anything else
-   * should return `undefined` and will be handled as normal.
-   *
-   * For example if `coder.com` is specified `8080.coder.com` will be proxied
-   * but `8080.test.coder.com` and `test.8080.coder.com` will not.
-   *
-   * Throw an error if proxying but the user isn't authenticated.
-   */
-  public maybeProxy(route: ProviderRoute, request: http.IncomingMessage): HttpResponse | undefined {
-    // Split into parts.
-    const host = request.headers.host || ""
-    const idx = host.indexOf(":")
-    const domain = idx !== -1 ? host.substring(0, idx) : host
-    const parts = domain.split(".")
-
-    // There must be an exact match.
-    const port = parts.shift()
-    const proxyDomain = parts.join(".")
-    if (!port || !this.options.proxyDomains.includes(proxyDomain)) {
-      return undefined
-    }
-
-    // Must be authenticated to use the proxy.
-    route.provider.ensureAuthenticated(request)
-
-    return {
-      proxy: {
-        port,
-      },
-    }
+        return wrapped
+      }),
+    ])
+    return this
+  }
+  // Overwrite `get` so we can distinguish between websocket and non-websocket
+  // routes. While we're at it handle async responses.
+  ;(express.Router as any).get = function get(route: expressCore.PathParams, ...handlers: express.Handler[]) {
+    originalGet.apply(this, [
```

Comment:
Unfortunately `this` will be undefined, I think because `originalGet` is from the `Router()` prototype and not the actual router.
```
    var route = this.route(path)
                     ^
TypeError: Cannot read property 'route' of undefined
```